pkg/sql/catalog/descriptor.go

```diff
@@ -505,6 +507,9 @@ type TableDescriptor interface {
 	// CheckConstraintUsesColumn returns whether the check constraint uses the
 	// specified column.
 	CheckConstraintUsesColumn(cc *descpb.TableDescriptor_CheckConstraint, colID descpb.ColumnID) (bool, error)
+	// IsShardColumn returns true if col corresponds to a non-dropped hash sharded
+	// index. This method assumes that col is currently a member of desc.
+	IsShardColumn(col Column) bool
```

Comment:
This wasn't defined in the TableDescriptor interface for some reason, only on tabledesc.Mutable.

Reply:
It's because a column's "shardedness" is defined in the table's indexes. Encoding this property in `Column` itself would be possible, but not really worth it in my opinion. Right now `Column` is backed by a struct which wraps a `*descpb.ColumnDescriptor` and a few metadata (column desc is in a mutation or not, that kind of thing). Adding a "shardedness" property would mean reengineering that backing struct a bit, which I'm keen to defer on doing until it's really necessary.